packages/cli/src/databases/utils/migrationHelpers.ts

```diff
@@ -39,30 +38,41 @@ export function loadSurveyFromDisk(): string | null {
 	}
 }
 
-let logFinishTimeout: NodeJS.Timeout;
+let runningMigrations = false;
 
-export function logMigrationStart(migrationName: string, disableLogging = inTest): void {
-	if (disableLogging) return;
+function logMigrationStart(migrationName: string): void {
+	if (inTest) return;
 
-	if (!logFinishTimeout) {
+	if (!runningMigrations) {
 		logger.warn('Migrations in progress, please do NOT stop the process.');
+		runningMigrations = true;
 	}
 
 	logger.debug(`Starting migration ${migrationName}`);
-
-	clearTimeout(logFinishTimeout);
 }
 
-export function logMigrationEnd(migrationName: string, disableLogging = inTest): void {
-	if (disableLogging) return;
+function logMigrationEnd(migrationName: string): void {
+	if (inTest) return;
 
 	logger.debug(`Finished migration ${migrationName}`);
-
-	logFinishTimeout = setTimeout(() => {
-		logger.warn('Migrations finished.');
-	}, 100);
 }
 
+const runDisablingForeignKeys = async (
+	migration: BaseMigration,
+	context: MigrationContext,
+	fn: MigrationFn,
+) => {
+	const { dbType, queryRunner } = context;
+	if (dbType !== 'sqlite') throw new Error('Disabling transactions only available in sqlite');
+	await queryRunner.query('PRAGMA foreign_keys=OFF');
+	await queryRunner.startTransaction();
+
+	await fn.call(migration, context);
+
+	await queryRunner.commitTransaction();
+	await queryRunner.query('PRAGMA foreign_keys=ON');
```

Comment:
What happens if we commit the transaction but then it fails on re-enabling the foreign keys? Asking in case we have problematic data, the transaction would be commited but then it would fail to enable foreign keys, which would lead the db in an inconsistent state forever. Do you think this is possible?

Reply:
good catch. we should do the re-enabling in a `finally` block. I'll update this.